src/config/watcher.rs

```diff
@@ -19,11 +22,57 @@ const CONFIG_WATCH_DELAY: std::time::Duration = std::time::Duration::from_secs(1
 
 const RETRY_TIMEOUT: std::time::Duration = std::time::Duration::from_secs(10);
 
+pub enum WatcherConfig {
+    /// recommended watcher for os, usually inotify for linux based systems
+    RecommendedWatcher,
+    /// poll based watcher. for watching files from NFS.
+    PollWatcher(u64),
+}
+
+enum Watcher {
+    /// recommended watcher for os, usually inotify for linux based systems
+    RecommendedWatcher(notify::RecommendedWatcher),
```

Comment:
Since the `Watcher` is necessary for use `watch` method on `PollWatcher` and `RecommendWatcher` . removing that makes `watch` method is not defined error

Reply:
I think you can `use notify::RecommendedWatcher` and `use notify::PollingWatcher`, that have nothing related to what you're saying about the `Watcher` trait. The previous code imported those two structs

and indeed, maybe you need to `use notify::Watcher` to use the `watch` method